tests/python/test_continous_dynamics.py

```diff
@@ -89,6 +89,140 @@ def test_multibody_free():
         pass
 
 
+def test_centroidal():
+    try:
+        space = manifolds.VectorSpace(9)
+        nk = 2
+        nu = 3 * nk
+        mass = 10.5
+        gravity = np.array([0, 0, -9.81])
+        contact_map = [(True, np.array([0, 0.1, 0])), (True, np.array([0.1, -0.1, 0]))]
+        ode = dynamics.CentroidalFwdDynamics(space, nk, mass, gravity, contact_map)
+        data = ode.createData()
+
+        assert isinstance(data, dynamics.CentroidalFwdData)
+
+        x0 = space.neutral()
+        u0 = np.random.randn(nu)
+
+        ode.forward(x0, u0, data)
+        ode.dForward(x0, u0, data)
+    except ImportError:
+        pass
+
+
+def test_centroidal_diff():
+    nx = 9
+    space = manifolds.VectorSpace(nx)
+    nk = 2
+    nu = 3 * nk
+    mass = 10.5
+    gravity = np.array([0, 0, -9.81])
+    contact_map = [(True, np.array([0, 0.1, 0])), (True, np.array([0.1, -0.1, 0]))]
+    ode = dynamics.CentroidalFwdDynamics(space, nk, mass, gravity, contact_map)
+    data = ode.createData()
+
+    x0 = np.random.randn(nx)
+    u0 = np.random.randn(nu)
+    epsilon = 1e-6
+
+    ode.forward(x0, u0, data)
+    ode.dForward(x0, u0, data)
+
+    xdot0 = data.xdot.copy()
+    Jx0 = data.Jx.copy()
+    Ju0 = data.Ju.copy()
+    Jxdiff = np.zeros((nx, nx))
+    Judiff = np.zeros((nx, nu))
+
+    for i in range(nx):
+        evec = np.zeros(nx)
+        evec[i] = epsilon
+        xi = x0 + evec
+        ode.forward(xi, u0, data)
+        ode.dForward(xi, u0, data)
+        Jxdiff[:, i] = (data.xdot - xdot0) / epsilon
+
+    for i in range(nu):
+        evec = np.zeros(nu)
+        evec[i] = epsilon
+        ui = u0 + evec
+        ode.forward(x0, ui, data)
+        ode.dForward(x0, ui, data)
+        Judiff[:, i] = (data.xdot - xdot0) / epsilon
```

Comment:
You can use finite difference helper instead of doing this (like you do for the residuals, it's possible in C++, I don't know if t's possible in Python).